include/blackhole/record.hpp

```diff
@@ -15,12 +15,12 @@ using cpp17::string_view;
 
 class record_t {
 public:
+    struct inner_t;
```

Comment:
Why do require this to be public, as you only additionally use it in private ctor?